security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/DistributedSecurityAnalysis.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2018, All partners of the iTesla project (http://www.itesla-project.eu/consortium)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.distributed;
+
+import com.powsybl.computation.*;
+import com.powsybl.contingency.ContingenciesProvider;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.xml.NetworkXml;
+import com.powsybl.security.SecurityAnalysis;
+import com.powsybl.security.SecurityAnalysisParameters;
+import com.powsybl.security.SecurityAnalysisResult;
+import com.powsybl.security.SecurityAnalysisResultMerger;
+import com.powsybl.security.interceptors.SecurityAnalysisInterceptor;
+import com.powsybl.security.json.JsonSecurityAnalysisParameters;
+import com.powsybl.security.json.SecurityAnalysisResultDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * Security analysis implementation which distributes the work through X
+ * executions of the "itools security-analysis" command.
+ *
+ * @author Sylvain Leclerc <sylvain.leclerc@rte-france.com>
+ */
+public class DistributedSecurityAnalysis implements SecurityAnalysis {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DistributedSecurityAnalysis.class);
+
+    private final Network network;
+    private final ComputationManager computationManager;
+
+    public DistributedSecurityAnalysis(Network network, ComputationManager computationManager, List<String> extensions, int taskCount) {
+        this.network = network;
+        this.computationManager = computationManager;
+        this.extensions = extensions;
```

Comment:
It seems this parameters should not be null (NullPointerException can be thrown at line 127)